python/python/bystro/ancestry/inference.py

```diff
@@ -22,14 +25,13 @@
 logger = logging.getLogger(__name__)
 
 
-@dataclass(frozen=True)
-class AncestryModel:
+class AncestryModel(Struct, frozen=True, forbid_unknown_fields=True):
     """Bundle together PCA and RFC models for bookkeeping purposes."""
 
     pca_loadings_df: pd.DataFrame
     rfc: RandomForestClassifier
 
-    def __post_init__(self) -> "AncestryModel":
+    def __post_init__(self) -> None:
```

Comment:
By dropping attrs, we dramatically simplify our codebase. It is trivially easy to add runtime checks through __post_init__, as you would in data class.

I left the runtime validations checks even though they are entirely pointless, because I was not far enough along to understand whether there was some way that Pat could have been generating or accepting invalid probabilities. I've learned that isn't the case.

The only place where you need runtime validation, is when you are taking user data. So I will in a followup PR remove all __post__init validation, in favor of using msgspec's "Constraints" (https://jcristharif.com/msgspec/constraints.html), which will perform the requisite checks when deserializing messages. Notably, since all of the ancestry types are send only (they are sent to the backend, but never currently serialized and sent to the ancestry module), these constraints will never run, but maybe one day we'll want another module to take ancestry data in JSON form and check that it is correct.